src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -23,40 +23,163 @@ uct_tcp_sockcm_t *uct_tcp_sockcm_ep_get_cm(uct_tcp_sockcm_ep_t *cep)
                             super.iface);
 }
 
+static const char *uct_tcp_sockcm_cm_ep_peer_addr_str(uct_tcp_sockcm_ep_t *cep,
+                                                      char *buf, size_t max)
+{
+    struct sockaddr_storage remote_dev_addr = {0}; /* Suppress Clang false-positive */
+    socklen_t remote_dev_addr_len;
+    ucs_status_t status;
+
+    /* get the device address of the remote peer associated with the connected fd */
+    status = ucs_socket_getpeername(cep->fd, &remote_dev_addr, &remote_dev_addr_len);
+    if (status != UCS_OK) {
+        return NULL;
```

Comment:
is it valid to pass NULL pointer to "%s" format print? maybe print ucs_status_string(status) to buf?